subsys/bluetooth/services/cgms/cgms_socp.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/types.h>
+#include <string.h>
+#include <zephyr/zephyr.h>
+
+#include <bluetooth/services/cgms.h>
+#include "cgms_internal.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(cgms, CONFIG_BT_CGMS_LOG_LEVEL);
+
+/**@brief Specific Operation Control Point opcodes. */
+#define SOCP_OPCODE_RESERVED                          0x00
+#define SOCP_WRITE_CGM_COMMUNICATION_INTERVAL         0x01
+#define SOCP_READ_CGM_COMMUNICATION_INTERVAL          0x02
+#define SOCP_READ_CGM_COMMUNICATION_INTERVAL_RESPONSE 0x03
+#define SOCP_WRITE_GLUCOSE_CALIBRATION_VALUE          0x04
+#define SOCP_READ_GLUCOSE_CALIBRATION_VALUE           0x05
+#define SOCP_READ_GLUCOSE_CALIBRATION_VALUE_RESPONSE  0x06
+#define SOCP_WRITE_PATIENT_HIGH_ALERT_LEVEL           0x07
+#define SOCP_READ_PATIENT_HIGH_ALERT_LEVEL            0x08
+#define SOCP_READ_PATIENT_HIGH_ALERT_LEVEL_RESPONSE   0x09
+#define SOCP_WRITE_PATIENT_LOW_ALERT_LEVEL            0x0A
+#define SOCP_READ_PATIENT_LOW_ALERT_LEVEL             0x0B
+#define SOCP_READ_PATIENT_LOW_ALERT_LEVEL_RESPONSE    0x0C
+#define SOCP_SET_HYPO_ALERT_LEVEL                     0x0D
+#define SOCP_GET_HYPO_ALERT_LEVEL                     0x0E
+#define SOCP_HYPO_ALERT_LEVEL_RESPONSE                0x0F
+#define SOCP_SET_HYPER_ALERT_LEVEL                    0x10
+#define SOCP_GET_HYPER_ALERT_LEVEL                    0x11
+#define SOCP_HYPER_ALERT_LEVEL_RESPONSE               0x12
+#define SOCP_SET_RATE_OF_DECREASE_ALERT_LEVEL         0x13
+#define SOCP_GET_RATE_OF_DECREASE_ALERT_LEVEL         0x14
+#define SOCP_RATE_OF_DECREASE_ALERT_LEVEL_RESPONSE    0x15
+#define SOCP_SET_RATE_OF_INCREASE_ALERT_LEVEL         0x16
+#define SOCP_GET_RATE_OF_INCREASE_ALERT_LEVEL         0x17
+#define SOCP_RATE_OF_INCREASE_ALERT_LEVEL_RESPONSE    0x18
+#define SOCP_RESET_DEVICE_SPECIFIC_ALERT              0x19
+#define SOCP_START_THE_SESSION                        0x1A
+#define SOCP_STOP_THE_SESSION                         0x1B
+#define SOCP_RESPONSE_CODE                            0x1C
+
+#define SOCP_RSP_RESERVED_FOR_FUTURE_USE              0x00
+#define SOCP_RSP_SUCCESS                              0x01
+#define SOCP_RSP_OP_CODE_NOT_SUPPORTED                0x02
+#define SOCP_RSP_INVALID_OPERAND                      0x03
+#define SOCP_RSP_PROCEDURE_NOT_COMPLETED              0x04
+#define SOCP_RSP_OUT_OF_RANGE                         0x05
+
+extern uint16_t comm_interval;
```

Comment:
Don't use extern. Define a dedicated internal API that can be used to access this value from more than one source file

Reply:
I will encapsulate these variables that describe the state of the CGMS instance into a struct variable, and pass it to SOCP as an argument.